src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,173 @@
-        return null;
+    public T get(int index) throws IndexOutOfBoundsException {
+        if ((index >= size()) || (index<0)) {
+            throw new IndexOutOfBoundsException("Index dosn't much");
+
+        }
+        int i = 0;
+        MyLinkedList node = first;
+        while (i != index){
+            node = node.next;
+            i++;
+        }
+        return (T) node.item;
     }
 
     @Override
-    public T set(T value, int index) {
-        return null;
+    public T set(T value, int index) throws IndexOutOfBoundsException {
+        if ((index >= size()) || (index<0)) {
+            throw new IndexOutOfBoundsException("Index dosn't much");
+        }
+        int i = 0;
+        MyLinkedList node = first;
+        while (i != index){
+            node = node.next;
+            i++;
+        }
+        node.item = value;
+        return value;
     }
 
     @Override
-    public T remove(int index) {
-        return null;
+    public T remove(int index)  throws IndexOutOfBoundsException {
+        if ((index >= size()) || (index<0)) {
+            throw new IndexOutOfBoundsException("Index dosn't much");
+        }
+        int i = 0;
+        MyLinkedList node = first;
+        while (i != index){
+            node = node.next;
+            i++;
+        }
+        if (i == 0) {
+            node.next.prev = null;
+            node.first = node.next;
+            return (T) node.item;
+        }
+
+        if (node.next != null) {
+            node.next.prev = node.prev;
+            node.prev.next = node.next;
+        } else {
+            node.prev.next = null;
+            last = node.prev;
+        }
+        return (T) node.item;
     }
 
     @Override
     public boolean remove(T object) {
+        if (isEmpty()){
+            return false;
+        }
+        MyLinkedList node = first;
+        while (node != null){
+            if (node.item.equals(object) ){
+                if (node == first) {
+                    if (node.next != null) {
+                        node.next.prev = null;
+                        node.first = node.next;
+                    }
+                }
+
+                if (node.next != null) {
+                    node.next.prev = node.prev;
+                    node.prev.next = node.next;
+                } else {
+                    node.prev.next = null;
+                    node.last = node.prev;
+                }
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        if (isEmpty()){
+            return 0;
+        }
+        int i = 1;
+        MyLinkedList node = first;
+        while (node.next != null){
+            node = node.next;
+            i++;
+        }
+        return i;
     }
 
     @Override
     public boolean isEmpty() {
+        if (first == null) {
+            return true;
+        }
         return false;
```

Comment:
:) don't use if/else to return boolean
```suggestion
        return first == null;
```